deploy.py

```diff
@@ -166,38 +293,58 @@ def update_rustc_if_needed(self):
       # Need to update
       self.checked_command_output(
           ["rustup", "install", target_toolchain_fullstring])
-      self.checked_command_output(
-          ["rustup", "target", "add", "thumbv7em-none-eabi"])
+    targets = {x.arch for x in SUPPORTED_BOARDS.values()}
+    for arch in targets:
+      self.checked_command_output(["rustup", "target", "add", arch])
```

Comment:
Shouldn't we only update the selected board's arch?

Reply:
Done.
I started updating all because I wanted to build the app for all boards at one (the TAB file is supposed to handle that). But then I faced some limitations on what tockloader can do. We can revisit that once we can build a single TAB file.